### DISCUSSION TITLE
when i run the demo named "HelloWorld" on windows 7,  the console window cannot exit  after main function return.

### DISCUSSION THREAD

**Discussion by Chrisxujinhui** [eProsima/Fast-DDS#3123] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

the console window  exit  after main function return.

### Current behavior

the console window cannot exit  after main function return

### Steps to reproduce

run the demo on windows 7

### Fast DDS version/commit

2.5.0

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

**Comment by Mario-DL** [comment]
Hi @Chrisxujinhui,

According to our [CONTRIBUTING.md](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md) guidelines, Windows 7 is not an officially supported platform, so I am moving this issue to the corresponding discussion forum.
